browser/lib/transport/xhrrequest.js

```diff
@@ -122,19 +122,19 @@ var XHRRequest = (function() {
 		for(var h in headers)
 			xhr.setRequestHeader(h, headers[h]);
 
-		var onerror = xhr.onerror = function(err) {
-			err.code = 80000;
-			self.complete(err);
+		var errorHandler = function(errorEvent, message, code, statusCode) {
+			var errorMessage = message + ", errorEvent was " + Utils.inspect(errorEvent) + ", current statusText is " + self.xhr.statusText;
```

Comment:
> Also, have we checked that Utils.inspect() gives us something meaningful for the event?

In my tests (just aborting or cutting the connection) it hasn't contained anything useful, no (the stringify was only `{isTrusted: true}`, and the rest of the event object didn't have anything better). Left it in there in the hope that maybe there's some more interesting error might happen in the wild where (a) the event object included some new fields, and (b) they'd be included in the stringify. Perhaps not very likely, but I figure leaving it there doesn't cost much. Want me to remove it?